src/metaswitch/clearwater/config_manager/config_access.py

```diff
@@ -343,19 +353,48 @@ def main(args):
         try:
             upload_verified_config(config_loader,
                                    local_store,
-                                   args.config_type,
+                                   selected_config,
                                    args.force,
                                    args.autoconfirm)
         except (UserAbort, ConfigUploadFailed) as exc:
             log.error("Upload failed")
             sys.exit(exc)
 
 
+def lookup_config_type(config_type, config_location):
+    """
+    This takes the string defining the config_type from arg.parse and returns
+    a instance of the class containing data on the config type selected
+    """
+    config_classes = load_plugins_in_dir(PLUGIN_DIR, config_location)
+    for choice in config_classes:
+        if config_type == choice.name:
+            return choice
+        # The config_type will always be the choice.name of one class as
+        # parse_arguments will make sure config_type == choice.name
```

Comment:
Comments should always be placed above the code they reference. Also, try not to use pseudo-code in comments where possible :).